lib/rets/client.rb

```diff
@@ -203,10 +203,11 @@ def create_parts_from_response(response)
 
         return parts
       else
+        logger.debug "Rets::Client: Found 1 part (the whole body)"
+
         # fake a multipart for interface compatibility
         headers = {}
-        response.headers.each { |k,v| headers[k] = v[0] }
```

Comment:
There is double failure here:

* `response.header['content-type']` returns array, i.e. `['image/jpeg']` but `response.headers['content-type']` is a single object
* `response.header` is an hash-like class which automatically downcase `Content-Type`, `response.headers` is a regular `Hash` which keeps the key casing as-is.